### DISCUSSION TITLE
AI-based Organizing

### DISCUSSION THREAD

**Discussion by gowthamravichandran** [tfeldmann/organize#424] [Ideas]
First, thanks @tfeldmann for the wonderful work and sharing it w the community. I have been using Organize for more than a year now. 

There are a lot of usecases where the organizing the files has to be deterministic and can be achieved traditionally, but there are some cases where it is difficult to make deterministic decisions with traditional approaches. 

For example, 

**Usecase:** A word document from work that was downloaded should go into respective project's folder. 

**Challenge:** the document might not contain the name of the client or the project, but have references to overall context of the project and there is no easy deterministic way to achieve this. 

**Idea:** With the power of LLMs, we have the power to understand the document at previously unimaginable levels. With a little context from the user defined for each project, we can move the files to respective folders with ease. 

I am pretty sure the community would come up with even more creative use cases leveraging AI. 

I would love to hear more ideas and thoughts around implementation.

**Comment by tfeldmann** [comment]
Would be an absolute killer application. I‘d love that. It‘s kind of difficult coming up with a prompt that works for many usecases. In your example the LLM needs to know about your project structure?

A first step could be an ‘ai‘ filter which allows customizing the prompt. This should be easy to implement targeting local models (ollama).

**Reply by gowthamravichandran** [reply]
You are right, its difficult to come up with a prompt that works for many usecases. 

I was thinking along the lines of letting the user define the prompt, rather than us trying to handle it. (maybe we could provide some out of box options as well)

Instead of having one prompt handle many usecases, we can let each rule have a micro prompt defined by the user (essentially doing a classification problem) and handle file operations accordingly. 

For example:

Lets say, I download a word document or pdf like a medical document, and I want them stored in a specific folder. the rule can prompt to classify it as Medical document or not by an LLM model and then move it to that folder.

Let me know your thoughts. I apologize for not replying earlier. I somehow missed this notification.

**Reply by feather42** [reply]
I like the idea of an AI filter

**Comment by eleazar7** [comment]
This would be great! I found this project while searching the web for "AI File Automation" Apps.
